rust/src/table_state.rs

```diff
@@ -54,6 +54,15 @@ impl DeltaTableState {
         Ok(new_state)
     }
 
+    /// Construct a delta table state object from a list of actions
+    pub fn from_actions(actions: Vec<Action>) -> Result<Self, ApplyLogError> {
```

Comment:
We could inline this method and reuse it in the `from_commit` method.

```suggestion
    #[inline]
    pub fn from_actions(actions: Vec<Action>) -> Result<Self, ApplyLogError> {
```